build.sh

```diff
@@ -288,8 +289,29 @@ if hasArg clean; then
 fi
 if hasArg uninstall; then
     UNINSTALL=1
+
+    # If clean or uninstall targets given, run them prior to any other steps
+    if [[ "$INSTALL_PREFIX" != "" ]]; then
+        rm -rf ${INSTALL_PREFIX}/include/raft
+        rm -rf ${INSTALL_PREFIX}/include/raft_runtime
+        rm -f ${INSTALL_PREFIX}/lib/libraft_distance.so
+        rm -f ${INSTALL_PREFIX}/lib/libraft_nn.so
+        rm -rf ${INSTALL_PREFIX}/lib/cmake/raft
+    fi
+    # This may be redundant given the above, but can also be used in case
+    # there are other installed files outside of the locations above.
+    if [ -e ${LIBRAFT_BUILD_DIR}/install_manifest.txt ]; then
+        xargs rm -f < ${LIBRAFT_BUILD_DIR}/install_manifest.txt > /dev/null 2>&1
+    fi
+    # uninstall raft-dask and pylibraft installed from a prior "setup.py install"
+    # FIXME: if multiple versions of these packages are installed, this only
+    # removes the latest one and leaves the others installed. build.sh uninstall
+    # can be run multiple times to remove all of them, but that is not obvious.
+    pip uninstall -y raft-dask pylibraft
```

Comment:
It is unclear to me how we should parse `uninstall`.

Should we only run the `pip uninstall` if `PY_LIBRAFT_BUILD_DIR` points to a valid directory, or do you want to have uninstall remove a pip install version of either package no matter how it was gotten?

Reply:
Essentially, I have a user that has accidentally built and installed RAFT into their base conda enviroment and they just want to blow it away to start with a fresh install. I guess it's fair to add the restriction that `build.sh uninstall` should only uninstall artifacts which were built and installed previously using `build.sh`. In that case, I think maybe deleting everything from the manifest might be enough. If the `${PYLIBRAFT_BUILD_DIR}/install_manifest.txt` is enought to remove the package from any python package manager (conda, pip, or just a general non-conda python environment) then I think that would preferable. 

What are your thoughts?

Reply:
I mean- the `pip uninstall` is easy enough (and not specific to `build.sh`) that the user can just run that on their own if they want to remove all versions of raft from a python environment no matter how they were installed.

Reply:
> In that case, I think maybe deleting everything from the manifest might be enough. If the ${PYLIBRAFT_BUILD_DIR}/install_manifest.txt is enought to remove the package from any python package manager (conda, pip, or just a general non-conda python environment) then I think that would preferable.

@vyasr can correct me. I don't think that would be sufficient as scikit-build does some extra python logic around installation.

> pip uninstall -y raft-dask pylibraft

What happens on machines where pip doesn't exist or is unrelated to this `uninstall`? Like if we are uninstalling for `conda` won't this also potentially cause unintended side effects?

We might need more expressive uninstall commands `uninstall-conda`, `uninstall-pip` etc to remove ambiguity on what exactly the user wants removed

Reply:
I've changed the `uninstall` target to an flag (`--uninstall`) and made it conditionally uninstall the parts specified by the targets in the build.sh command (e.g. `./build.sh pylibraft --uninstall`) but the question does still remain whether this is the correct way to uninstall the python packages (@vyasr hopefully you have some insight and suggestions here). 

I'm not necessarily against having something like `--uninstall-lib`, `--uninstall-conda`, `--uninstall-pip`, though I almost wonder if the `conda` and `pip` uninstalls might be better documented in the build and install docs rather than doing them in the `build.sh`.  Looking for suggestions here.

Reply:
We coudl also do something like "if pip is available, uninstall using pip, otherwise, check to see if $CONDA_PREFIX is in the `_skbuild/install_manifest.txt` and if so, perform conda uninstall".

Reply:
I really like `./build.sh pylibraft --uninstall` as that makes the intent very clear to me